### ISSUE TITLE
Bundler v0.16.1 fails to locate bundle executable on Travis with Ruby 2.0.0 & Ruby 2.1.10

### ISSUE THREAD

**Issue by piotrmurach** [bundler/bundler#6245]
My [tty](https://github.com/piotrmurach/tty) works on top of `bundler` to provide scaffolding behaviour. My recent deployments on Ruby `2.0.0` and `2.1.10` failed on Travis CI due to failure to locate `bundle` executable. This failure occurred when bundler was upgraded from `v0.16.0` to the latest `v0.16.1`. An example of [failure backtrace](https://travis-ci.org/piotrmurach/tty/jobs/325208813): 
```
/home/travis/.rvm/gems/ruby-2.0.0-p648/bin/bundle:23:in `load': cannot load such file --
/home/travis/.rvm/rubies/ruby-2.0.0-p648/lib/ruby/gems/2.0.0/gems/bundler-1.16.1/exe/bundle (LoadError)
	from /home/travis/.rvm/gems/ruby-2.0.0-p648/bin/bundle:23:in `<main>'
	from /home/travis/.rvm/gems/ruby-2.0.0-p648/bin/ruby_executable_hooks:15:in `eval'
	from /home/travis/.rvm/gems/ruby-2.0.0-p648/bin/ruby_executable_hooks:15:in `<main>'
```
Downgrading fixes the issue.

**Comment by colby-swandale** [comment]
make sure you're on the latest version of rubygems

**Comment by colby-swandale** [comment]
what's the output of `bundle env`?

**Comment by yahonda** [comment]
It looks similar with https://github.com/rubygems/rubygems/issues/2123 . If your RubyGems version is 2.7.3, execute `gem update --system` to update RubyGems version to 2.7.4 and see if it resolves this error.

**Comment by colby-swandale** [comment]
@piotrmurach Did you resolve your issue? Looking at travis, your build history shows that you updated your `.travis.yml` and your builds are now passing.

**Comment by piotrmurach** [comment]
@yahonda Good to hear from you! You may not recall me, few years ago we had a drink in izakaya in Ueno district. Anyhow, thank you for the advice I will check. However, my thinking goes that the bundler should work with the rubygems preinstalled with default ruby installation for major Ruby releases. I'm not personally interested in a sense that I use latest Ruby myself. However, I'm building a tool that 'sits' on top of bundler executable and may blow in people faces if they use for example, Ruby 'v2.0.0'. Even if updating fixes the issue, it is still less than optimal.

@colby-swandale The test suite is fixed by downgrading to `v1.16.0`. To me it seems like a regression in terms of how bundler exec works. Is it possible that some newer features of rubygems are used? I would expect this kind of breakage at least in minor level but not in a patch upgrade.

**Comment by dalizard** [comment]
I am also experiencing the same (maybe) problem. Here's a bit more info:
- Ruby 2.4.0
- RubyGems 2.7.4
- Bundler 1.16.1, default: 1.16.0

_gem which bundler_ returns _~/.rubies/ruby-2.4.0/lib/ruby/site_ruby/2.4.0/bundler.rb_, while I think it should be returning _~/.gem/ruby/2.4.0/gems/bundler-1.16.1/lib/bundler.rb_.

**Comment by colby-swandale** [comment]
@dalizard please open a new issue, you follow our [ISSUE](https://github.com/bundler/bundler/blob/master/doc/contributing/ISSUES.md) for more information.

**Comment by davidkiser** [comment]
Just commenting that Im having the exact same issue on our environment with Ruby 2.3.5.  The path is looking down the 1.16.1 path, but the executable exists down 1.16.0

**Comment by perlun** [comment]
Seeing similar issues as @dalizard on Travis, with Ruby 2.5.0. In my case, setting `RUBYOPT=''` when executing `bundle exec foo` from an already-bundled scenario helped workaround the issue.

**Comment by colby-swandale** [comment]
This issue was fixed in #6689 which was released in Bundler 1.16.5
